met-api/src/met_api/services/authorization.py

```diff
@@ -2,9 +2,41 @@
 
 This module is to handle authorization related queries.
 """
+from flask_restx import abort
+
+from met_api.constants.membership_type import MembershipType
+from met_api.models.membership import Membership as MembershipModel
+from met_api.models.user import User as UserModel
+from met_api.utils.user_context import UserContext, user_context
 
 
 # pylint: disable=unused-argument
+@user_context
 def check_auth(**kwargs):
     """Check if user is authorized to perform action on the service."""
-    return True
+    user_from_context: UserContext = kwargs['user_context']
+    token_roles = user_from_context.roles
+    permitted_roles = kwargs.get('one_of_roles', [])
+    has_valid_roles = bool(set(token_roles) & set(permitted_roles))
+    if has_valid_roles:
+        return
+    if MembershipType.TEAM_MEMBER.name in permitted_roles:
```

Comment:
A bit confused about line 21-22 and then line 23. If user has valid roles we return? Then how do we reach 23?

Reply:
if they have a valid role (say admin, or super admin) , we dont need to check membership because he is already authorised ..